src/components/chatbot/ChatBox.tsx

```diff
@@ -0,0 +1,64 @@
+'use client'
+
+import React, { useEffect, useRef, useState } from 'react'
+import ChatBoxForm, { formSchema } from './ChatBoxForm';
+import ChatMessage from './ChatMessage';
+import { useChat } from '@/hooks/useChat';
+import { useSendMessage } from '@/hooks/useBot';
+import { ByUser, Message } from '@/types/messages/message';
+import { getMessagesFromChat } from '@/lib/chat';
+import { getMessageWebsocketUrl } from '@/constants/global';
+import { z } from 'zod';
+import { useWebsocket } from '@/hooks/useWebsocket';
+
+type P = {
+  // ID of the chat to retrieve history of messages from. Not the TG Chat ID, but the actual chat Id stored in database
+  chatId: number;
+}
+
+export default function ChatBox({chatId}: P) {
+  const {data: chat, isPending} = useChat(chatId);
+  const sendMessageMutation = useSendMessage(chatId)
+
+  const [messages, setMessages] = useState<Message[]>([])
+  const bottomRef = useRef<any>(null);
+  
+  useWebsocket(getMessageWebsocketUrl(), setMessages)
+
+  useEffect(() => {
+    if (chat) {
+      setMessages(getMessagesFromChat(chat));
+    }
+  }, [chat])
+
+  // Auto-scrol to bottom of chat upon new messages
+  useEffect(() => {
+    bottomRef.current?.scrollIntoView({ behavior: "smooth" });
+  }, [messages])
```

Comment:
True, I wasn't thinking about this case. My main motivation for adding this code is because when the user sends a message, it doesn't automatically scroll to the bottom. Would it be better if I just made it so it scrolled to the bottom only when the user sends a message? And maybe play a sound when they receive a message in a future PR? As you said, this is no longer in the scope of the ticket

Reply:
> My main motivation for adding this code is because when the user sends a message, it doesn't automatically scroll to the bottom. Would it be better if I just made it so it scrolled to the bottom only when the user sends a message?

My concern is exactly the fact that scrolling occurred without user input. (Making decisions on behalf of the user).

I think I might be missing the context. Is the problem here that when a new message is updated, React renders that message below the current bottom message (which is hidden)?

If thats the case, I agree with the autoscrolling, but only if the user isnt currently trying to scroll up and look through the chat history